### PR TITLE
Add factory, publish to npm and auto test

### PR DESCRIPTION
**Pull request by malteish**
**There are no changes to the DssVest smart contract in this PR!**

Instead, multiple updates to increase the repository's safety and utility, as well as adding the factory contract.
The changes are:
- add factory contract to allow automatic deployment of new DssVestMintable smart contract by calling a function on-chain
- add hardhat and config to publish an npm package
- enable github actions and enforce successfull foundry and echidna tests before merging to master branch
- update docs

Please pay special attention to [the factory contract](https://github.com/corpus-io/dss-vest/pull/15/files#diff-9e1d585bd1b9f2044bdae415807e12be1588dd527a043b657f51cad74fbc85e7).

### CODE REVIEW

File: src/Factory.sol [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+pragma solidity 0.8.17;
+
+import {DssVestMintable} from "./DssVest.sol";
+
+contract DssVestNaiveFactory {
+
+    function createDssVestMintable(address forwarder, address companyToken, address companyAdminAddress) external returns (address) {
```
Comment (CJentzsch) [inline src/Factory.sol:9]: Looks good. But I still would feel better if we could find an existing audited smart contract that can be used for deployments.

File: certora/DSToken.sol [hunk 2]
```diff
@@ -214,7 +214,7 @@ contract DSToken is DSMath, DSAuth {
     }
 
     function approve(address guy) external returns (bool) {
-        return approve(guy, uint(-1));
+        return approve(guy, type(uint256).max);
```
Comment (CJentzsch) [inline certora/DSToken.sol:217]: Why do we need to make changes to the certora contracts?

**Review by CJentzsch — COMMENTED** [review]
I would rather keep the certora contracts as is, even though they are using an older solidity compiler version.

File: src/Factory.sol (hunk 1 above)
Comment (malteish) [reply src/Factory.sol:9]: I totally agree. But I think such a contract does not exist, because each instance needs these specific arguments on creation.

**Review by CJentzsch — APPROVED** [review]

**Pull request merged**